src/PluggableAuth.php

```diff
@@ -75,6 +89,15 @@ public function authenticate( &$id, &$username, &$realname, &$email, &$errorMess
 			return false;
 		}
 
+		/* this is a feature after updating wikis which used strtolower on usernames.
+		 * to use it, set this in LocalSettings.php:
+		 * $LDAPAuthenticationUsernameNormalizer = 'strtolower';
+		 */
+		$normalizer = $config->get( "UsernameNormalizer" );
+		if ( !empty( $normalizer ) && function_exists( $normalizer ) ) {
+			$username = ucfirst( call_user_func( $normalizer, $username ) );
```

Comment:
* ucfirst isn't needed and should probably be avoided.
* will this work with closures?

Reply:
You are right, ucfirst() is not needed.
If function_exists() is replaced by is_callable() then closures work, too. i would not recommend using this, but it works.
I tried it with:
$LDAPAuthenticationUsernameNormalizer = function($name) {return 'AD'.$name;};

Reply:
Why do you not recommend this?

Reply:
Being able to enter more than simple functions might seduce the one who configurates this plugin to enter less maintenable code. That i would not recommend that is only a personal opinion, thanks for merging :)